addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1455,6 +1490,118 @@ export class ListRenderer extends Component {
             this.onCreateAction(context);
         }
     }
+   /**
+     * Create a list of fields for the property values of the 'properties' field
+     * @param {*} baseFieldObject The 'properties' field
+     * @returns a list of field for the property values
+     */
+    getPropertyFieldsFromPropertiesField(baseFieldObject) {
+        const fields = [];
+        for(const property of this.getPropertyDefinitions()) {
+            const propertyName = this.getPropertyFieldName(property.name, baseFieldObject.name);
+            const propertyColumn = {
+                ...baseFieldObject,
+                FieldComponent: PropertyValue,
+                rawAttrs: {
+                    ...baseFieldObject.rawAttrs,
+                    name: propertyName,
+                },
+                name: propertyName,
+                label: property.string,
+                propertyType: property.type,
+                string: property.string,
+            };
+            fields.push(propertyColumn)
+        }
+        return fields;
+    }
+
+    /**
+     * Create columns for properties
+     * @param {Array} columns all the columns (typically props.archInfo.columns)
+     * @returns the same array with the properties columns replaced by 1 column per property
+     */
+    propertiesGetAllColumns(columns) {
+        columns = [...columns];
+        for(const propertyName of this.getPropertiesFieldNames()) {
+            const propertiesColumnIndex = columns.findIndex(column => column.name === propertyName);
+            if(propertiesColumnIndex >= 0) {
+                const baseColumnObject = columns[propertiesColumnIndex];
+                const propertyColumns = this.getPropertyFieldsFromPropertiesField(baseColumnObject);
+                let property_column_id = 1;
+                for(const column of propertyColumns){
+                    column.id = `${column.id}_property_${property_column_id}`;
+                    property_column_id++;
+                }
```

Comment:
```suggestion
                for(let i = 1; i <= propertyColumns.length; ++i){
                    propertyColumns[i].id = `${column.id}_property_${i}`;
                }
```

a bit simpler imo (sad that JS doesn't have `enumerate` like python)